### DISCUSSION TITLE
[Feature Request]  Documentation on testing web components with typescript

### DISCUSSION THREAD

**Discussion by emmanuelagarry** [open-wc/open-wc#2015] [Q&A] [answered]
Hey guys, I have been trying to set up a testing environment for my projects that uses lit-element and typescript, but there is very little documentation on how to achieve that with the open-wc testing libraries. Is there any other good alternative to testing web components that has good documentation for typescript

**Comment by LarsDenBakker** [comment, marked as answer]
Typescript compiles to javascript, so if you compile to JS you can follow any of the existing documentation how to how to write tests. We have documentation for the test runner here https://modern-web.dev/guides/test-runner/getting-started/, and some docs on how to use typescript with the dev server which powers the test runner: https://modern-web.dev/guides/dev-server/typescript-and-jsx/

**Reply by emmanuelagarry** [reply]
Okay, it works. Thanks for the help.  I was a bit concerned at first that I had to build my files all the time, but the building process is ridiculously fast 😲.

**Reply by emmanuelagarry** [reply]
Turns out I don't need to build it myself to run the tests., The documentation actually has everything; I just needed to read it repeatedly to understand 😅
